qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,90 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base class for calibration-type experiments."""
+
+from abc import abstractmethod
+from typing import Optional
+
+from qiskit.providers.options import Options
+from qiskit.providers.backend import Backend
+
+from qiskit_experiments.framework.base_experiment import BaseExperiment
+from qiskit_experiments.framework.experiment_data import ExperimentData
+
+
+class BaseCalibrationExperiment(BaseExperiment):
+    """An abstract base class for calibration experiments.
+
+    This abstract base class specifies an experiment and how to update an
+    optional instance of :class:`Calibrations` specified in the experiment options
+    under calibrations. Furthermore, the experiment options also specifies
+    an auto_update variable which, by default, is set to True. If this variable,
+    is True then the run method of the experiment will call :meth:`block_for_results`
+    and update the calibrations instance.
+    """
+
+    # The updater class that updates the Calibrations instance
+    __updater__ = None
+
+    @abstractmethod
+    def update_calibrations(self, experiment_data: ExperimentData):
+        """Update parameter values in the :class:`Calibrations` instance.
+
+        Subclasses must implement this method which will call the :meth:`update`
+        method of the updater.
+        """
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default options for experiment
+
+        Experiment Options:
+            calibrations (Calibrations): An optional instance of :class:`Calibrations` if this
```

Comment:
The problem is this docstring is not propagated to the subclass. I think I can upgrade autodoc to support nested default experiment options in separate PR.

Reply:
That sounds good.